ADGR/ Overview frontend

eadams17 opened the pull request:
## WHAT
build out the architecture needed to support the frontend of the Admin Growth Diagnostic Reports and add the scaffolding for the Overview tab

## WHY
this is the first step in getting the frontend ready for the new ADGR!

## HOW
- some refactoring to handle the routing of the reports as well as abstracting some shared logic
- adding a parent `DiagnosticGrowthReportsContainer` to house the shared logic and a `OverviewTable` that will render the overview specific data

### Screenshots
(If applicable. Also, please censor any sensitive data)
<img width="1423" alt="Screen Shot 2023-10-10 at 1 00 19 PM" src="https://github.com/empirical-org/Empirical-Core/assets/25959584/d3db2e62-0f92-4517-875d-ba66ba239f14">

### Notion Card Links
(Please provide links to any relevant Notion card(s) relevant to this PR.)
https://www.notion.so/quill/New-Project-Card-Admin-Diagnostic-Reports-Spec-For-Three-Reports-15ace9cefb7c4bf7b5475b041fff56d9?d=d6bcd80e0df94fe5902f127b4f789bae#f1801ca73f2d4b98825d7f42e30797ef

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  yes
Have you deployed to Staging? | yes
Self-Review: Have you done an initial self-review of the code below on Github? | yes
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? | yes

eadams17 committed:
add header for diagnostic reports overview

eadams17 committed:
Merge branch 'develop' of https://github.com/empirical-org/Empirical-Core into admin-diagnostic-report/overview-fe

eadams17 committed:
add icons to buttons

eadams17 committed:
add tabs, dropdown and table

eadams17 committed:
consolidate CSS and reconcile with develop

emilia-friedberg committed:
update fill in the blanks concept results file to use shared function (#11109)

emilia-friedberg committed:
replace diagnostic-grey icon (#11110)

eadams17 committed:
add more dropdown options

eadams17 committed:
tweak CSS

eadams17 committed:
update active states for buttons

eadams17 committed:
add tests

eadams17 committed:
fix linter errors

eadams17 committed:
update snapshot

cissyxyu committed:
Change styling and copy on Diagnostic Reports pages (#11104)

* change to recommended practice

* try appending description

* try another way of stacking paragraphs

* try another way

* add one more break for visual matching

* snapshot update

* Style changes for shifted bigger element on progress page

* Reverse tooltip changes

* particularize style changes to only affect the cards in post diagnstic

* add card class

cissyxyu committed:
Re-style Empty Progress Reports page (#11107)

* html and css changed

* make text bolder on link

cissyxyu committed:
Update CMS ruby version to Ruby 3.2.0 (#11108)

* update gemfile and ruby versions

* update circleCI version

brendanshean committed:
Add rake task for updating missing labels (#11113)

* Add rake task for updating missing labels

* Activate model after update

* Add some logging

brendanshean reviewed (approved):
Nice work on this!

emilia-friedberg commented on services/QuillLMS/app/assets/stylesheets/pages/premium_hub/growth-diagnostic-reports.scss:
```diff
@@ -0,0 +1,74 @@
+.diagnostic-reports-main {
+  max-width: calc(100% - 317px) !important;
```
Maybe throw in a comment about where the 317px is coming from?

eadams17 replied:
I'll add a comment

emilia-friedberg commented on services/QuillLMS/client/app/bundles/PremiumHub/containers/PremiumFilterableReportsContainer.tsx:
```diff
@@ -290,8 +289,11 @@ export const PremiumFilterableReportsContainer = ({ accessType, adminInfo, locat
       />
-      {shouldRenderDataExportContainer && <DataExportContainer {...sharedProps} />}
-      {shouldRenderUsageSnapshotsContainer && <UsageSnapshotsContainer {...sharedProps} />}
+      <Switch>
```
I'm excited that we're able to use react-router here now! Any idea what changed? Also, it would be good to use [react-router v6 syntax](https://reactrouter.com/en/main/upgrading/v5) here if possible.

eadams17 replied:
I think it was an issue with using `component` rather than `render`

emilia-friedberg commented on services/QuillLMS/client/app/bundles/PremiumHub/containers/diagnosticGrowthReports/index.tsx:
```diff
@@ -0,0 +1,130 @@
+      <div className="header">
+        <h1>
+          <span>Diagnostic Growth Report</span>
+          <a href="" rel="noopener noreferrer" target="_blank">
```
I'm assuming we just don't have the link for this yet?

eadams17 replied:
correct-- no link yet

emilia-friedberg commented on services/QuillLMS/client/app/bundles/PremiumHub/containers/diagnosticGrowthReports/overviewTable.tsx:
```diff
@@ -0,0 +1,100 @@
+    return <Spinner />
+  }
+
+  return (
+    <DataTable
+      className="growth-diagnostic-reports-overview-table reporting-format"
+      emptyStateMessage={noResultsMessage('diagnostic')}
+      headers={headers}
+      rows={[]}
+    />
+  )
```
This is just empty right now because we haven't got the backend hooked up, right?

eadams17 replied:
correct-- @anathomical is currently working on the backend, but he has been keeping the Notion doc up-to-date with expected payload shape

emilia-friedberg reviewed (commented):
Looks great to me overall! One little thing about react-router and a couple of tiny things to confirm but very excited about where this project is right now.

anathomical committed:
Lms/admin snapshot front end retry (#11103)

* Testing in staging with 1 second retry

* WIP add some testing console logs to understand how this is working

* More testing

* WIP more testing

* Hopefully-final version of the code

* Add some logging

* Add more logging

* More logging

* Even more logging

* Pass retryTimeout through to getData

* More logging, trying to figure out encapsulation

* Allow for set calls on useState values to be async

* Hopefully final update with working clearTimeout

* Still trying to figure out clearTimeout

* Still chasing down timing issue

* Remove final logging

* Remember to update Rankings to retry with the same logic

anathomical committed:
Lms/send pusher in sidekiq job (#11105)

* Create a Sidekiq worker to send Pusher messages

So that we can de-couple Pusher errors and retries from the rest of jobs that utilize Pusher

* Refactor Admin Snapshot workers to use Pusher Sidekiq worker

Instead of sending messages directly

* Refactor classroom import workers to use Pusher Sidekiq job

Instead of sending Pusher messages directly from the import workers.  This lets us decouple errors and retries triggered by Pusher from the actual work of the importers

* Fix linting error

eadams17 committed:
Merge branch 'develop' of https://github.com/empirical-org/Empirical-Core into admin-diagnostic-report/overview-fe

eadams17 committed:
add CSS comments

eadams17 committed:
fix linter
